snapcraft/commands/daemon.start

```diff
@@ -620,7 +620,8 @@ fi
 # LXD
 ## Check for existing LXDs
 for pid in $(pgrep -f "lxd --logfile" ; pgrep -f "lxd.debug --logfile"); do
-    grep -qF "SNAP_NAME=lxd" "/proc/${pid}/environ" || continue
+    # Cheap confirmation that we likely have a LXD PID
+    grep -qxFm1 "SNAP_NAME=lxd" "/proc/${pid}/environ" || continue
```

Comment:
Are there long form arguments we can use here? Either that or an extension of the commit message/comment to explain how this is an improvement/cheaper?

Reply:
While they all have long form versions, I think switching them all would make it harder to understand to some as most of those options I've only ever seen used in their short form.

I'm assuming the `-m1` bit is the one needing clarification here which I'm happy to expand as indeed it's not commonly used and the long form is quite clear (`--max-count=1`):
>        -m NUM, --max-count=NUM
>              Stop  reading  a  file  after  NUM matching lines.

Reply:
but the use of --max-count makes sense to me now

Reply:
I've now expanded `-x` and `-F` as well.